tools/update-semver.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/sh
+
+# Shell script to update semver to the latest release
+
+set -ex
+
+cd "$( dirname "$0" )/.." || exit
+
+rm -rf deps/semver
+
+rm -rf semver-tmp
+mkdir semver-tmp
+
+(
+    cd semver-tmp || exit
+    
+    ROOT="$PWD/.."
+    [ -z "$NODE" ] && NODE="$ROOT/out/Release/node"
+    [ -x "$NODE" ] || NODE=$(command -v node)
+    NPM="$ROOT/deps/npm/bin/npm-cli.js"
+    NPX="$ROOT/deps/npm/bin/npx-cli.js"
+
+    # Get latest release from npm
+    "$NODE" "$NPM" init --yes
+    "$NODE" "$NPM" install --global-style --no-bin-links --ignore-scripts semver
+
+    cd "$ROOT"
+
+    # Move the latest version into deps
+    mv semver-tmp/node_modules/semver deps
+    rm -rf semver-tmp
+
+    # Bundle source into single file
+    cd "deps/semver"
+    "$NODE" "$NPX" esbuild@0.14.38 index.js --bundle --platform=node --outfile=semver.js
```

Comment:
Why does this require `semver` to be downloaded again? Why can't we run the `esbuild` step on `deps/npm/node_modules/semver`?

Reply:
Semver is only downloaded once, the lines you highlighted just cd into the cloned directory and run esbuild on it

Reply:
I mean why do we need to download the source of `semver` when it already exists inside `deps/npm/node_modules/semver`.

Reply:
It makes sense to me to use only the version of semver that npm brings in

Reply:
> Should probably put the updating semver script in the npm update script as well then

Most npm PRs are automated and created with tooling from the `npm/cli` repo. Is there a step that could run after one of those is merged, that would look for `semver` and then bundle its source?

Also there is a chance that `semver` gets deduped in the npm tree in the future. Ideally the tool could use something like `npm query` to get only the version of semver that npm uses.

When run inside the `npm/cli` repo:
```
❯ npm query ':root > #semver' | jq '.[0].location'
"node_modules/semver"
```

Reply:
> Do you mean the part of the script that generates the single file?

Yeah

Reply:
I assume semver doesn't need to be added to dep_checker anymore now that it'll be vendored through whatever version npm is using?

> Is there a step that could run after one of those is merged, that would look for `semver` and then bundle its source?

Are you referring to an action that would fire when a pr is merged with npm cli or something different?

> ❯ npm query ':root > #semver' | jq '.[0].location'

Currently not able to get this working for whatever reason but that would indeed be a good idea as well